api/index.js

```diff
@@ -7,25 +7,39 @@ const apiRequest = (
   url,
   method = 'GET',
   queryParams = {},
-  customHeaders = {}
+  customHeaders = {},
+  data
 ) => {
-  const controller = new AbortController()
-  const { signal } = controller
-
+  // Initialise URL
   const requestUrl = new URL(/^\w+:\/\//.test(url) ? url : `${API_URL}${url}`)
-  const requestHeaders = {
-    Accept: 'application/json',
-    ...customHeaders,
-  }
-
   requestUrl.search = new URLSearchParams(queryParams).toString()
 
-  const fetchPromise = fetch(requestUrl, {
+  // Initialise options
+  const requestOptions = {
     method,
     credentials: 'include',
-    headers: requestHeaders,
-    signal,
-  })
+    headers: {
+      Accept: 'application/json',
+    },
+  }
+
+  // Process body
+  if (method.toUpperCase() !== 'GET') {
+    if (typeof data == 'object') {
```

Comment:
```suggestion
    if (typeof data === 'object') {
```

Reply:
We use `eqeqeq: smart` for a reason. `typeof` always returns a string, so if you compare it to another string, there is no need to write another character.